core/src/main/java/org/pentaho/reporting/platform/plugin/output/util/ExportReportUtils.java

```diff
@@ -0,0 +1,196 @@
+import org.pentaho.reporting.libraries.formula.EvaluationException;
+import org.pentaho.reporting.libraries.formula.parser.ParseException;
+
+import java.util.Date;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public interface ExportReportUtils {
+  String FILTERS_SUMMARY = "Filters Summary";
+  String PROMPTS_SUMMARY = "Prompts Summary";
+  String NO_FILTERS = "No filters used";
+  String NO_PROMPTS = "No prompts used";
+
+  // ISO_LOCAL_DATE is yyyy-MM-dd
+  DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ISO_LOCAL_DATE;
+
+  Map<Class<?>, Function<Object, String>> typeFormattingStrategies = Map.of(
+    String.class, String.class::cast,
+    Number.class, Object::toString,
+    Date.class, ExportReportUtils::formatDate,
+    String[].class, value -> String.join( ", ", (String[]) value ),
+    Number[].class, value -> formatNumberArray( (Number[]) value ),
+    Date[].class, value -> formatDateArray( (Date[]) value )
+  );
+
+  private static String formatDate( Object value ) {
+    if ( value instanceof java.sql.Date ) {
+      // java.sql.Date has toLocalDate()
+      return ( (java.sql.Date) value ).toLocalDate().format( DATE_FORMATTER );
+    } else if ( value instanceof java.util.Date ) {
+      // Convert util.Date to Instant, then to LocalDate
+      Instant instant = ( (java.util.Date) value ).toInstant();
+      LocalDate localDate = instant.atZone( ZoneId.systemDefault() ).toLocalDate();
+      return localDate.format( DATE_FORMATTER );
+    } else {
+      throw new IllegalArgumentException( "Not a Date: " + value );
+    }
+  }
+
+  private static String formatNumberArray( Number[] array ) {
+    return Arrays.stream( array )
+      .map( Object::toString )
+      .collect( Collectors.joining( ", " ) );
+  }
+
+  private static String formatDateArray( Date[] dates ) {
+    return Arrays.stream( dates )
+      .map( ExportReportUtils::formatDate )
+      .collect( Collectors.joining( ", " ) );
+  }
+
+  /*
+   * Formats the parameter value based on its type.
+   *
+   * @param parameterValue The parameter value to format.
+   * @return The formatted string representation of the parameter value.
+   */
+  static String formatParameterValue( Object parameterValue ) {
+    return typeFormattingStrategies.entrySet().stream()
+      .filter( entry -> entry.getKey().isInstance( parameterValue ) )
+      .findFirst()
+      .map( entry -> entry.getValue().apply( parameterValue ) )
+      .orElseThrow( () -> new IllegalArgumentException(
+        "Unsupported type: " + parameterValue.getClass().getSimpleName()
+      ) );
+  }
+
+  default String getReadableFilterDescription() {
+    return null;
+  }
+
+  @VisibleForTesting
+  default void addElementToReportHeader( Element element, ReportHeader reportHeader ) {
```

Comment:
Extract and implement in the Impl class

Reply:
Please see my response here:  https://github.com/pentaho/pentaho-platform-plugin-reporting/pull/899#discussion_r2136964765